### PR TITLE
PKI Responses Part 4

### PR DESCRIPTION
**Pull request by AnPucel**
Responses for files: `path_resign_crls`, `path_revoke`, `path_roles`, `path_root`, `path_sign_issuers`, and `path_tidy`

Please see this [RFC](https://github.com/hashicorp/vault/pull/18626) for implementation details.

Related PRs:
- [PKI Config](https://github.com/hashicorp/vault/pull/18376)
- [PKI Fetch](https://github.com/hashicorp/vault/pull/18596)
- [PKI Issue & Manage](https://github.com/hashicorp/vault/pull/18479)

### CODE REVIEW

File: builtin/logical/pki/path_tidy.go
Comment (dhuckins) [inline builtin/logical/pki/path_tidy.go:330]: same int here
https://github.com/hashicorp/vault/blob/14e5fe0695df5acf63f6be048d808b9166332e2a/builtin/logical/pki/path_tidy.go#L966
```suggestion
							"issuer_safety_buffer": {
								Type:        framework.TypeInt,
								Description: `Issuer safety buffer`,
								Required:    true,
							},
```

File: builtin/logical/pki/path_tidy.go
Comment (dhuckins) [inline builtin/logical/pki/path_tidy.go:355]: looks like an int
https://github.com/hashicorp/vault/blob/14e5fe0695df5acf63f6be048d808b9166332e2a/builtin/logical/pki/path_tidy.go#L1031
```suggestion
							"interval_duration": {
								Type:        framework.TypeInt,
								Description: `Specifies the duration between automatic tidy operation`,
								Required:    true,
							},
```

File: builtin/logical/pki/path_tidy.go
Comment (dhuckins) [inline builtin/logical/pki/path_tidy.go:380]: looks like an int
https://github.com/hashicorp/vault/blob/14e5fe0695df5acf63f6be048d808b9166332e2a/builtin/logical/pki/path_tidy.go#L1036

```suggestion
							"safety_buffer": {
								Type:        framework.TypeInt,
								Description: `Safety buffer time duration`,
								Required:    true,
							},
```

File: builtin/logical/pki/path_revoke.go
Comment (AnPucel) [reply builtin/logical/pki/path_revoke.go:36]: okay! Thanks. I'll have to double check my other PRs. Pretty sure I marked this as true for most.

File: builtin/logical/pki/path_roles.go
Comment (AnPucel) [reply builtin/logical/pki/path_roles.go:59]: hmmm. I think I might've copy/pastaed from the input and didn't see these extra fields. Thanks!

File: builtin/logical/pki/path_roles.go
Comment (AnPucel) [reply builtin/logical/pki/path_roles.go:824]: yeppp. I was looking at the function beneath it (Doh)

File: builtin/logical/pki/path_tidy.go
Comment (AnPucel) [reply builtin/logical/pki/path_tidy.go]: Gotcha! I missed that. Thanks

File: builtin/logical/pki/path_tidy.go
Comment (AnPucel) [reply builtin/logical/pki/path_tidy.go:296]: ahh thanks.

File: builtin/logical/pki/path_tidy.go
Comment (AnPucel) [reply builtin/logical/pki/path_tidy.go:72]: hmm. yeah. Probably just mark them as optional, but I'll let @averche weigh in.

**Comment by averche** [comment]
For the time or duration related fields please consider using either `framework.TypeTime` or `framework.TypeDurationSeconds` as these will translate better in the generated code.

https://github.com/hashicorp/vault/blob/40f599b11cc164804d0cabfdb20664dff84bccee/sdk/framework/openapi.go#L805-L807

https://github.com/hashicorp/vault/blob/40f599b11cc164804d0cabfdb20664dff84bccee/sdk/framework/openapi.go#L785-L787

File: builtin/logical/pki/path_revoke.go
Comment (averche) [inline builtin/logical/pki/path_revoke.go:36]:
```suggestion
								Required:    true,
```
Should this be Required: true?

File: builtin/logical/pki/path_roles.go
Comment (averche) [inline builtin/logical/pki/path_roles.go:34]:
```suggestion
								Required:    true,
```

File: builtin/logical/pki/path_revoke.go
Comment (averche) [inline builtin/logical/pki/path_revoke.go:251]:
```suggestion
								Required:    true,
```

File: builtin/logical/pki/path_revoke.go
Comment (averche) [inline builtin/logical/pki/path_revoke.go:105]: I did not see this one in the code, could you please double-check.

File: builtin/logical/pki/path_revoke.go
Comment (averche) [inline builtin/logical/pki/path_revoke.go:159]: Same as above.

File: builtin/logical/pki/path_tidy.go
Comment (averche) [reply builtin/logical/pki/path_tidy.go:124]: Agreed, probably mark these as optional

**Review by averche — COMMENTED** [review]
A few suggestions regarding Required/Optional, otherwise LGTM.

For the `"keys"` field, typically all the “LIST” request return “keys” in every successful code path (hence Required: true). If that doesn’t apply, feel free to ignore those.

File: builtin/logical/pki/path_revoke.go
Comment (AnPucel) [reply builtin/logical/pki/path_revoke.go:105]: It looks like this can get added [here](https://github.com/hashicorp/vault/blob/8651d6de38890ae5d74903d21971386b915b81e5/builtin/logical/pki/path_revoke.go#LL470C23-L470C23) 

and adds that `state`

Comment (AnPucel) [reply builtin/logical/pki/path_revoke.go:105]: it also goes through this path and can get added [here](https://github.com/hashicorp/vault/blob/8651d6de38890ae5d74903d21971386b915b81e5/builtin/logical/pki/crl_util.go#L1023)

Comment (cipherboy) [reply builtin/logical/pki/path_revoke.go:105]: @AnPucel is correct :-)

Comment (averche) [reply builtin/logical/pki/path_revoke.go:105]: Ah completely missed that code path, thanks for checking!

File: builtin/logical/pki/path_fetch_issuers.go
Comment (averche) [inline builtin/logical/pki/path_fetch_issuers.go]: I believe these are dummy fields that will not show up in the final response, we should probably remove them from the schema.

Comment (AnPucel) [reply builtin/logical/pki/path_fetch_issuers.go]: So, not having these ends up breaking some of the tests. b/c the responses can return either these fields or the other responses. Specifically it was breaking ones with the new callback. 

It does end up in the response that's getting returned to the test in this case.

Comment (dhuckins) [reply builtin/logical/pki/path_fetch_issuers.go]: this one is a bit confusing, seems like the function can return non-json 
 https://github.com/hashicorp/vault/blob/29406c43a157d3d6541c4ed03a655ee28dcd08e8/builtin/logical/pki/path_fetch_issuers.go#L1174-L1178
so even though 1 function handles multiple paths, it can return different content types/response shapes

Comment (AnPucel) [reply builtin/logical/pki/path_fetch_issuers.go]: Removed these.

**Review by averche — APPROVED** [review]
LGTM, thanks for adding these!

**Pull request merged**
